### ISSUE TITLE
Autoclose puts extra ) when I type (

### ISSUE THREAD

**Issue by ricpelo** [geany/geany-plugins#319]
Using geany plugins 1.26, from Ubuntu packages:

When I type (, Geany shows ()). The same for [, { and <.

**Comment by codebrainz** [comment]
I don't suppose you also have the builtin auto-closing on? (Edit->Preferences->Editor->Completions->Auto-close quotes and brackets).

**Comment by ricpelo** [comment]
@codebrainz no, that's the first thing I checked. The builtin autoclosing is off for all type of brackets (, {, [, ' and ".

**Comment by scriptum** [comment]
@ricpelo could you try to compile geany-plugins yourself? I had same issue with older versions. Binaries compiled by Ubuntu maintainers had this bug by unknown reason. Possibly this is due to build flags/environment.

**Comment by chamfay** [comment]
This happen to me also, i'm using Geany 1.27 and i compiled version 1.28 also
but the same problem,
any idea please?

**Comment by scriptum** [comment]
Looks like something in Ubuntu build configuration breaks plugin. While compiling plugin from scratch I cannot reproduce this.

I asked maintainers and didn't get response.

**Comment by b4n** [comment]
If build options break the plugin, I would rather put my money in a bug in the plugin (relying on undefined behavior, or at best, implementation-defined behavior) like memory corruption or something like this, that some protective build flags prevent from happening.

I'd suggest trying to run the plugin under i.e. Valgrind, and/or with some address sanitizer to see if they can't find some problems.  I might try that later today, but I'm not sure I'd know how to reproduce at least the steps if not the behavior.

**Comment by dmaphy** [comment]
I'm observing the same behaviour here with different braces and quotes (with current 1.29!). In my opinion it's quite obviously that the feature from auto-close plugin is conflicting with the Geany builtin features.

At this point I don't understand what exactly the auto-close plugin is doing different than Geanys builtin features, but I guess most likely there is a difference? Why would someone re-implement this feature as a plugin if there wasn't?

However, the plugin should check if Geanys builtin feature is ticked for the following and then disable (grey out) the tickboxes for parenthesis, curly brackets, square brackets, single quotes and double quotes. To manually tick and untick is…well…you have to move from keyboard to your mouse, you know…

**Comment by Akronix** [comment]
I cannot reproduce it with Geany 1.32 (I'm using the [ppa Geany repo](https://launchpad.net/~geany-dev))

**Comment by klausmcm** [comment]
I am able to produce this issue in geany 1.29 (installed from Debian Stretch repository). Autoclosing of quotes (single and double) doesn't work either.
What's odd is that I have the same Debian running in a virtual machine and geany is the same version and the issues are not there.

**Comment by elextr** [comment]
@klausmcm as stated above it seems to be an issue with Debian, not Geany, and 1.29 is pretty old now, best to upgrade.

**Comment by klausmcm** [comment]
I like to stick with whatever comes with Debian even if a bit old.
I compared my virtual machine and regular machine installation and noticed that the difference is in ibus. I have ibus installed for typing Korean characters. Closing ibus actually seemed to fix the issue so the issue is somewhere between Geany and ibus. I ended up replacing ibus with scim. No more problems for now.

**Comment by sipiatti** [comment]
@klausmcm thank you! I have two workstations and a work-laptop, but only on one of them the problem occured. I remember that there was an ibus update on it...

**Comment by LarsGit223** [comment]
@ricpelo: can this issue be closed? It does not seem to be related to Geany itself - see all the comments above. I cannot re-produce the issue with current Geany on Ubuntu 18.04.3.
